tests/test_util.fish

```diff
@@ -37,19 +37,21 @@ if not set -q __fish_is_running_tests
     end
     set -l IFS # clear IFS so cmd substitution doesn't split
     cd (dirname $script); or die
-    set -xl XDG_CONFIG_HOME (printf '%s/tmp.config/%s' $PWD %self); or die
-    if test -d $XDG_CONFIG_HOME
-        # if the dir already exists, we've reused a pid
-        # this would be surprising to reuse a fish pid that failed in the past,
-        # but clear it anyway
-        rm -r $XDG_CONFIG_HOME; or die
-    end
+
+    set -lx XDG_DATA_HOME ../test_data
```

Comment:
The original implementation was overly complicated. There is absolutely no need for a directory name that changes with each test run. Surely no one is running multiple `make test` invocations in parallel. And if you do so the results won't be predictable because too many of the tests create files with invariant names in /tmp.